crates/interpreter/src/opcode/tables.rs

```diff
@@ -0,0 +1,179 @@
+        Self::Plain(make_instruction_table::<H, SPEC>())
+    }
+}
+
+impl<'a, H: Host + ?Sized + 'a> InstructionTables<'a, H> {
+    /// Inserts the instruction into the table with the specified index.
+    #[inline]
+    pub fn insert(&mut self, opcode: u8, instruction: Instruction<H>) {
+        match self {
+            Self::Plain(table) => table[opcode as usize] = instruction,
+            Self::Boxed(table) => table[opcode as usize] = Box::new(instruction),
+        }
+    }
+
+    /// Converts the current instruction table to a boxed variant if it is not already, and returns
+    /// a mutable reference to the boxed table.
+    #[inline]
+    pub fn to_boxed(&mut self) -> &mut BoxedInstructionTable<'a, H> {
+        self.to_boxed_with(|i| Box::new(i))
+    }
+
+    /// Converts the current instruction table to a boxed variant if it is not already with `f`,
+    /// and returns a mutable reference to the boxed table.
+    #[inline]
+    pub fn to_boxed_with<F>(&mut self, f: F) -> &mut BoxedInstructionTable<'a, H>
+    where
+        F: FnMut(Instruction<H>) -> BoxedInstruction<'a, H>,
+    {
+        match self {
+            Self::Plain(_) => self.to_boxed_with_slow(f),
+            Self::Boxed(boxed) => boxed,
+        }
+    }
+
+    #[cold]
+    fn to_boxed_with_slow<F>(&mut self, f: F) -> &mut BoxedInstructionTable<'a, H>
+    where
+        F: FnMut(Instruction<H>) -> BoxedInstruction<'a, H>,
+    {
+        let Self::Plain(table) = self else {
+            unreachable!()
+        };
+        *self = Self::Boxed(make_boxed_instruction_table(table, f));
+        let Self::Boxed(boxed) = self else {
+            unreachable!()
+        };
+        boxed
+    }
+
+    /// Returns a mutable reference to the boxed instruction at the specified index.
+    #[inline]
+    pub fn get_boxed(&mut self, opcode: u8) -> &mut BoxedInstruction<'a, H> {
+        &mut self.to_boxed()[opcode as usize]
+    }
+
+    /// Inserts a boxed instruction into the table at the specified index.
+    #[inline]
+    pub fn insert_boxed(&mut self, opcode: u8, instruction: BoxedInstruction<'a, H>) {
+        *self.get_boxed(opcode) = instruction;
+    }
+
+    /// Replaces a boxed instruction into the table at the specified index, returning the previous
+    /// instruction.
+    #[inline]
+    pub fn replace_boxed(
+        &mut self,
+        opcode: u8,
+        instruction: BoxedInstruction<'a, H>,
+    ) -> BoxedInstruction<'a, H> {
+        core::mem::replace(self.get_boxed(opcode), instruction)
+    }
+
+    /// Replaces a single instruction in the table at the specified index with `f`.
+    #[inline]
+    pub fn update_boxed<F>(&mut self, opcode: u8, f: F)
+    where
+        F: Fn(&DynInstruction<'a, H>, &mut Interpreter, &mut H) + 'a,
+    {
+        update_boxed_instruction(self.get_boxed(opcode), f)
+    }
+
+    /// Replaces every instruction in the table by calling `f`.
+    #[inline]
+    pub fn update_all<F>(&mut self, f: F)
```

Comment:
```suggestion
    pub fn replace_all<F>(&mut self, f: F)
```

Reply:
Im using the same convention as in std::panic::*hook, its not exactly replace because you are taking the previous one as input, so its updating

Reply:
Scratch this, update is better as we are taking old instruction function